docs/enhancements/add-external-providers.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Add-Providers-Support
+authors:
+  - "@Gal-Zaidman"
+creation-date: 2021-07-25
+last-updated: 2021-07-25
+---
+
+# Add Providers Support
+
+## Summary
+
+Currently, the assisted installer supports Bare-Metal installations only.
+Openshift can be installed on various providers such as:
+- Red Hat Virtualization.
+- OpenStack.
+- vSphere.
+- AWS.
+- GCP.
+- Azure.
+
+This enhancement will define a clear interface to extend the assisted installer and allow
+provider installations.
+
+## Motivation
+
+Assisted Installer can be a great tool to perform UPI (User Provisioned) installation on external providers,
+providing the user with clear UI to see its provisioned machines joining the cluster, and lowering installation
+resources requirements.
+
+Assisted Installer is also ideal for mixed env configurations.
+** The RHV provider supports installing a mixed env where the Control-Plain (Master) nodes are deployed on top of RHV
+and Compute (Worker) nodes are Bare-Metal nodes.
+
+### Goals
+
+- Create a Provider interface, and hooks in the assisted installer code which separates the assisted installer core code
+ from the provider-specific code.
+- Add clear documentation for each provider specific function.
+- Document steps to install a provider cluster with assisted installer
+
+### Non-Goals
+
+- Support IPI installations.
+- Integrating all relevant [Openshift installer types](https://github.com/openshift/installer/tree/master/pkg/types)
+into the assisted installer code - this will require a large refactor of the assisted installer code.
+- Remove Baremetal from not appropriate struct/function names (example in open questions).
+ 
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a RHV admin I want to create an Openshift cluster on my RHV engine by using the assisted installer.
+
+#### Story 2
+
+As a RHV admin I want to create an Openshift cluster on my RHV engine by using the assisted installer in which
+Control-Plain (Master) nodes are deployed on top of RHV and Compute (Worker) nodes are Bare-Metal nodes.
+
+#### Story 3
+
+As an Openshift developer working on platform X I want to add support for installing Openshift with assisted installer
+on top of platform X.
```

Comment:
This is something that will be possible when this work is done. However, I wonder if we should make it an explicit part of the stories supporting this enhancement. There's a discussion with regards to only supporting `on-prem` and not cloud providers but, as a developer reading this enhancement, I may feel encouraged to add support for cloud providers too.

Am I missing something?

Reply:
The whole point of creating this generic infrastructure is to make it easier to support more platforms.  I guess that's the point of this story.

Reply:
We talked about it above, I don't think it encourages you to implement cloud platforms, but just doesn't talk about it.
The interface which we will define is about adding a provider, there is probably nothing to specific for cloud provider but adding any provider will require the approval and support of the assisted installer team.
We do mention adding cloud providers as a non-goal of the current enhancement if it helps